app/services/document_capture_session_result.rb

```diff
@@ -24,7 +24,11 @@ def selfie_status
     self[:selfie_status].to_sym
   end
 
-  alias_method :success?, :success
+  def success_status
```

Comment:
Commenting inline so it's a threaded conversation.

Questions:

- I think that  `:success?` is what determines whether the response is a 200 or 400?
- Is `successful_result?` basically a replacement for `all_passed?` , but excluding the selfie check?
- I’m seeing this in the A/Cs but not reflected in the code. Are the A/Cs wrong or the code?
    - first attempt to read from DocumentCaptureSessionResult's selfie_success and doc_auth_success fields. If that does not succeed, the code will then read from the older success field.
    

I think the non-refactor changes are:

- Change `doc_auth_success?` to check `attention_with_barcode?` and `id_type_supported?`
- Add `success_status` which is the selfie and doc status combined
- Change the `success?` alias to point to `success_status`

Reply:
> Commenting inline so it's a threaded conversation.
> 
> Questions:
> 
> * I think that  `:success?` is what determines whether the response is a 200 or 400?

Nope, this is the **result**, used to be indicating full success. 

> * Is `successful_result?` basically a replacement for `all_passed?` , but excluding the selfie check?
Are you talking about successful_result? in true_id_response? If so, not exactly.

> * I’m seeing this in the A/Cs but not reflected in the code. Are the A/Cs wrong or the code?

This is a little bit complicated, because there are changes introduced by ongoing works. Let me think and see whether i can explain or make it more explicit.

>   
>   * first attempt to read from DocumentCaptureSessionResult's selfie_success and doc_auth_success fields. If that does not succeed, the code will then read from the older success field.

Reply:
> Commenting inline so it's a threaded conversation.
> 
> Questions:
> 

>   
>   * first attempt to read from DocumentCaptureSessionResult's selfie_success and doc_auth_success fields. If that does not succeed, the code will then read from the older success field.
> 
This is not explicitly listed in AC, but in description, I think it's cautious to do both, so modified the `successful_status` method.

Combed through the code the `success` filed is accessed through original `success?` method in two places: [link_sent_controller](https://github.com/18F/identity-idp/blob/848a9633a23f99ae618bb97b020d1f1e30fa4fdf/app/controllers/idv/link_sent_controller.rb#L82-L82) and [capture_doc_status_controller](https://github.com/18F/identity-idp/blob/848a9633a23f99ae618bb97b020d1f1e30fa4fdf/app/controllers/idv/capture_doc_status_controller.rb#L36-L36)